WordPress/Classes/ViewRelated/What's New/Data store/Cache/AnnouncementsCache.swift

```diff
@@ -0,0 +1,60 @@
+import WordPressKit
+
+/// Generic feature announcement cache
+protocol AnnouncementsCache {
+    var announcements: [Announcement]? { get set }
+}
+
+
+/// UserDefaults-based cache for feature announcements
+struct UserDefaultsAnnouncementsCache: AnnouncementsCache {
+
+    var announcements: [Announcement]? {
+        get {
+            guard let announcements = UserDefaults.standard.announcements, versionIsValid(for: announcements) else {
+                    UserDefaults.standard.announcements = nil
+                    return nil
+            }
+            return announcements
+        }
+        set {
+            UserDefaults.standard.announcements = newValue
+        }
+    }
+
+    private func versionIsValid(for announcements: [Announcement]) -> Bool {
+        guard let minimumVersion = announcements.first?.minimumAppVersion, // there should not be more than one announcement
```

Comment:
It might be good to have a couple of tests for this code.

Reply:
Sure, as the description says, Unit Tests will be added in a separate PR.

Reply:
Sorry, missed that! :)